onnxruntime/python/tools/transformers/onnx_exporter.py

```diff
@@ -255,34 +306,48 @@ def export_onnx_model(model_name, opset_version, use_external_data_format, model
-                                                  False)
+    return validate_and_optimize_onnx(model_name, use_external_data_format, model_type, onnx_dir, input_names, use_gpu,
+                                      precision, optimize_onnx, validate_onnx, use_raw_attention_mask, overwrite, config,
+                                      model_fusion_statistics, onnx_model_path, example_inputs, example_outputs_flatten,
+                                      tokenizer)
 
-    if optimize_onnx or precision == Precision.FLOAT16 or precision == Precision.INT8:  # Use script (optimizer.py) to optimize
-        optimized_model_path = get_onnx_file_path(onnx_dir, model_name, len(input_names), True, use_gpu, precision,
-                                                  False, use_external_data_format)
-        optimize_onnx_model(onnx_model_path, optimized_model_path, model_type, config.num_attention_heads,
-                            config.hidden_size, use_gpu, precision, use_raw_attention_mask, overwrite,
-                            model_fusion_statistics)
 
-        onnx_model_path = optimized_model_path
-        if validate_onnx:
-            is_valid_onnx_model = validate_onnx_model(onnx_model_path, example_inputs, example_outputs_flatten, use_gpu,
-                                                      precision == Precision.FLOAT16)
+def export_onnx_model_from_tf(model_name, opset_version, use_external_data_format, model_type, model_class, cache_dir, onnx_dir,
+                              input_names, use_gpu, precision, optimize_onnx, validate_onnx, use_raw_attention_mask, overwrite,
+                              model_fusion_statistics):
 
-        if precision == Precision.INT8:
-            logger.info(f"Quantizing model: {onnx_model_path}")
-            QuantizeHelper.quantize_onnx_model(onnx_model_path, onnx_model_path)
-            logger.info(f"Finished quantizing model: {onnx_model_path}")
+    config = AutoConfig.from_pretrained(model_name, cache_dir=cache_dir)
 
-    else:  # Use OnnxRuntime to optimize
-        if is_valid_onnx_model:
-            ort_model_path = get_onnx_file_path(onnx_dir, model_name, len(input_names), False, use_gpu, precision, True,
-                                                use_external_data_format)
-            optimize_onnx_model_by_ort(onnx_model_path, ort_model_path, use_gpu, overwrite, model_fusion_statistics)
+    model = load_pretrained_model(model_name, config=config, cache_dir=cache_dir, custom_model_class=model_class, if_tf_model=True)
 
-    max_input_size = tokenizer.max_model_input_sizes[model_name] if model_name in tokenizer.max_model_input_sizes else 1024
+    model._saved_model_inputs_spec = None
+
+    tokenizer = AutoTokenizer.from_pretrained(model_name, cache_dir=cache_dir)
+    example_inputs = tokenizer.encode_plus("This is a sample input", return_tensors="tf", max_length=512, pad_to_max_length=True, truncation=True)
```

Comment:
How about using a config attribute (like max query length) to replace 512. Different model might have different settings.